Added documentation for TM1638 Led and Key module

skykingjwc opened the pull request:
## Description:

Documentation for new TM1638 Component

**Related issue (if applicable):** fixes <link to issue>

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3340

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

skykingjwc committed:
Create tm1638.rst

Created tm1638.rst copied from tm1637.rst as template.  Will alter with new info for TM1638 component

skykingjwc committed:
Update tm1638.rst

Removed referenced to 1637, added sections for LED and buttons.  Will need to provide additional docs in future commit.

skykingjwc committed:
Add files via upload

Added image for TM1638

skykingjwc committed:
Delete TM1638-full.JPG

skykingjwc committed:
Add files via upload

Upload image for TM1638

skykingjwc committed:
Delete tm1638-full.JPG

skykingjwc committed:
Add files via upload

skykingjwc committed:
Update tm1638.rst

Added code block sections for LED and buttons.  Will need to replace content.

skykingjwc committed:
Update tm1638.rst

skykingjwc committed:
Update tm1638.rst

Corrected component names

skykingjwc committed:
Update tm1638.rst

Corrected typo

skykingjwc committed:
Update tm1638.rst

Added correct pins to clock example

skykingjwc committed:
Update tm1638.rst

Corrected verbiage for the buttons

skykingjwc committed:
Update tm1638.rst

Corrected typo in component name for image label

skykingjwc committed:
Update index.rst

Added link for TM1638

skykingjwc committed:
Update tm1638.rst

Fixed indenting

skykingjwc committed:
Add files via upload

Added image for TM1638

skykingjwc committed:
Update tm1638.rst

Fixed underline

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+        on_press:
+         then:
+            - switch.turn_on: MyLed
+        on_relase
```
`    on_release:`

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+
+Buttons
+----------------
+The TM1638 7 Segment Display Keypad & LED Module has 8 buttons across the top or middle of the display, depending on the board.  These buttons are exposed as binary sensor components with the key property set between 0 and 7,  They can be accessed via the binary sensor component configuration.  As with the LEDs the buttons are labeled 1-8 on the board, but use zero based addressing.  Therefore th ...
```
`(link)`?

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+
+LEDs
+----------------
+The TM1638 7 Segment Display Keypad & LED Module has 8 LED lights across the top or middle of the display, depending on the board.  These LEDs are exposed as switches with the led property set between 0 and 7.  The board has the LEDs labled 1-8, but the leds use zero based addressing.   The LED labled "1" is LED 0 in the configuraiton, so on and so forth.
```
You could add a link to the `switch` docs here, probably on `switches`.

skykingjwc committed:
Update tm1638.rst

Fixed typo for on_release

skykingjwc committed:
Update tm1638.rst

skykingjwc committed:
Corrected spelling errors and shortened LED explanation

skykingjwc committed:
Corrected spelling error

skykingjwc committed:
Another spelling error

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+- **clk_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The pin you have the CLK line hooked up to.
+- **dio_pin** (**Required**, :ref:`Pin Schema <config-pin_schema>`): The pin you have the DIO line hooked up to.
+- **intensity** (*Optional*, int): The intensity with which the TM1638 should drive the outputs. Range is from
+  0 (least intense) to 7 (the default).  This setting will affect the 7 segment display and the 8 indivdual LED lights.
```
individual

skykingjwc replied:
Corrected spelling error

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+7-segment Display Keypad & LED Module does not have a concept of individual pixels. In the lambda you're passed a variable called ``it``
+as with all other displays. In this case however, ``it`` is a TM1638 instance (see API Reference).
+
+The most basic operation with the TM1638 is wiring a simple number to the screen as in the configuration example
```
wiring -> writing

skykingjwc replied:
Corrected spelling error

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+            it.strftime("%H%M", id(homeassistant_time).now());
+
+
+current/components/switch/index.rst
```
accidental text?

skykingjwc replied:
Removed unnecessary text

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+current/components/switch/index.rst
+LEDs
+----------------
+The TM1638 7 Segment Display Keypad & LED Module has 8 LED lights across the top or middle of the display, depending on the board.  These LEDs are exposed as switches (:doc:`Switch </components/switch>`) with the led property set between 0 and 7.  The board has the LEDs labelled 1-8, but the leds use zero based addressing.   The LED labelled "1" is LED 0 in the configuration.
```
:doc:\`switches </components/switch/index>\`

skykingjwc replied:
Added doc link

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+
+Buttons
+----------------
+The TM1638 7 Segment Display Keypad & LED Module has 8 buttons across the top or middle of the display, depending on the board.  These buttons are exposed as binary sensor components with the key property set between 0 and 7,  They can be accessed via the binary sensor component configuration.  As with the LEDs the buttons are labeled 1-8 on the board, but use zero based addressing.  Therefore th ...
```
See :doc:\`/components/binary_sensor/index\`

skykingjwc replied:
Added doc link

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,168 @@
+See Also
+--------
+
+- :doc:`index`
```
Add the doc links for display, switch, and binary_sensor here as well.

skykingjwc replied:
Added doc links

skykingjwc committed:
Corrected spelling errors and added doc links

ssieb commented on components/display/tm1638.rst:
```diff
@@ -126,10 +126,9 @@ The following example creates a typical digital clock with the ``:`` colon flash
 LEDs
 ----------------
-The TM1638 7 Segment Display Keypad & LED Module has 8 LED lights across the top or middle of the display, depending on the board.  These LEDs are exposed as switches (:doc:`Switch </components/switch>`) with the led property set between 0 and 7.  The board has the LEDs labelled 1-8, but the leds use zero based addressing.   The LED labelled "1" is LED 0 in the configuration.
+The TM1638 7 Segment Display Keypad & LED Module has 8 LED lights across the top or middle of the display, depending on the board.  These LEDs are exposed as switches (:doc:`switches </components/switch/index>`) with the led property set between 0 and 7.  The board has the LEDs labelled 1-8, but the leds use zero based addressing.   The LED labelled "1" is LED 0 in the configuration.
```
You misunderstood what I meant there.  The doc link replaces the `switches` word and remove the parentheses.  Otherwise, move the doc link to the end like the other section.

skykingjwc replied:
Yea I saw it, my bad :)

skykingjwc replied:
Parens removed

ssieb commented on components/display/tm1638.rst:
```diff
@@ -164,5 +163,7 @@ See Also
 --------
 
 - :doc:`index`
+- :doc:`switches </components/switch/index>`
```
The first part gives the text to use for the link.  In this case, you don't need that. So just:
:doc:\`/components/switch/index\`

skykingjwc replied:
Removed unnecessary word 'switches'

skykingjwc committed:
Removed duplicate word

skykingjwc committed:
Removed parens

skykingjwc committed:
Added documentation for using outputs and switches at the same time to control LEDs

The LEDs can be exposed as outputs or switches.  Added documentation to clarify this.

skykingjwc committed:
Added switch component section header

skykingjwc committed:
Fixed apostrophes

ssieb commented on components/display/tm1638.rst:
```diff
@@ -0,0 +1,189 @@
+         then:
+            - switch.turn_off: TM1638Led1
+            
+    switch:            
```
This isn't correct.  I assume this was supposed to be a second binary_sensor.

skykingjwc replied:
Yep,  corrected it.  Thanks for the catch.

skykingjwc committed:
Removed incorrect switch, corrected spelling error

ssieb committed:
Merge branch 'next' into next
